src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,74 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static final Map<Class<?>, Class<?>> implementaionMap = new HashMap<>();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
+        initializationMap(implementaionMap);
```

Comment:
Use a verb in method naming. Also it is no need to pass a map to method. 
One more point. If you map is private static final, it is constant and it's content can be initialized as constant.
```suggestion
        initializeMap();
```